protocol/contracts/libraries/LibFunction.sol

```diff
@@ -72,22 +72,30 @@ library LibFunction {
      @return data The function call return datas
     **/
     function buildAdvancedCalldata(
-        bytes calldata callData,
-        bytes calldata advancedData,
+        bytes memory callData,
+        bytes memory advancedData,
```

Comment:
It would be great to leave this as calldata to keep the gas costs of the standard farm version to a minimum. Do you have any thoughts on how to achieve this? Maybe add a `buildAdvancedCalldataMem` version?